lib/salesforce/api.rb

```diff
@@ -80,6 +80,9 @@ def set_access_token
       if status == 400 && body.include?("user hasn't approved this consumer")
         AdminDashboardData.add_problem_message(APP_NOT_APPROVED)
       end
+      if status >= 300 && SiteSetting.salesforce_api_error_logs
```

Comment:
Is this going to fire again on the `status == 400 && body.include?("user hasn't approved this consumer")` case? Is that desired?

Reply:
Nope, just checking that the error logging wasn't overly broad.

Reply:
Something that I _just_ realised (thanks to actually seeing that error message) is that the user probably needs to just approve the app... 😅 (or it's the wrong user in the settings! https://github.com/discourse/discourse-salesforce/blob/b6eb5ed5438b6d6336c2b167a537b4b6eadba2c2/lib/salesforce/api.rb#L97-L105)